packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts

```diff
@@ -242,6 +254,45 @@ export class StatefulDagLMDGHOST implements ILMDGHOST {
     return Buffer.from(this.justified.bestTarget.blockRoot, "hex");
   }
 
+  // To address the bouncing attack, only update conflicting justified
+  //  checkpoints in the fork choice if in the early slots of the epoch.
+  public shouldUpdateJustifiedCheckpoint(blockRoot: Hash): boolean {
+    if(!this.justified) {
+      return true;
+    }
+    if(this.computeSlotsSinceEpochStart(getCurrentSlot(this.config, this.genesisTime)) < 
+       this.config.params.SAFE_SLOTS_TO_UPDATE_JUSTIFIED) {
+      return true;
+    }
+    const newJustifiedBlock = this.nodes[blockRoot.toString("hex")];
+    if (newJustifiedBlock.slot <= this.justified.slot) {
+      return false;
+    }
+    if (this.getAncestor(blockRoot.toString("hex"), this.justified.slot) !== this.justified.blockRoot) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private computeSlotsSinceEpochStart(slot: Slot): number {
```

Comment:
I think this method should go in state transition package if it's not already there. You could also have option param epcoh

Reply:
yeah it's done